cpp/include/raft/neighbors/detail/cagra/search_multi_cta.cuh

```diff
@@ -569,7 +546,7 @@ struct search : public search_plan_impl<DATA_T, INDEX_T, DISTANCE_T> {
   ~search() {}
 
   void operator()(raft::resources const& res,
-                  raft::device_matrix_view<const DATA_T, INDEX_T, row_major> dataset,
+                  raft::device_matrix_view<const DATA_T, INDEX_T, layout_stride> dataset,
```

Comment:
Would there be any benefit to using a  [padded layout](https://github.com/rapidsai/raft/blob/567dfd74a5c092cec61de6b52ea239ea91b2ed46/cpp/include/raft/core/mdspan.hpp#L58) here or having an overload for it in the public API just to simplify the conversion?